### PR TITLE
when operator collection does not exists in workspace

### PR DESCRIPTION
**Pull request by MariandreaMV**
addresing issue https://github.com/IBM/operator-collection-sdk-vscode-extension/issues/114

### CODE REVIEW

**Review by freemanlatrell — CHANGES_REQUESTED** [review]
@MariandreaMV Can the dropdown below be updated to use the `select` dropdown, giving the user only the `Yes` or `No` options to select. Here's an example https://github.com/IBM/operator-collection-sdk-vscode-extension/blob/main/src/utilities/util.ts#L245-L250
image

File: package.json [hunk 1]
```diff
@@ -225,12 +225,17 @@
...
       },
       {
         "view": "operator-collection-sdk.operators",
         "contents": "Select a valid project in Openshift and create a ZosCloudBroker instance to use this extension",
-        "when": "operator-collection-sdk.loggedIn && !operator-collection-sdk.zosCloudBrokerInstalled && !operator-collection-sdk.validNamespace"
+        "when": "operator-collection-sdk.loggedIn && !operator-collection-sdk.zosCloudBrokerInstalled && !operator-collection-sdk.validNamespace &&operator-collection-sdk.isCollectionInWorkspace"
+      },
+      {
+        "view": "operator-collection-sdk.operators",
+        "contents": "Operator Collection isn’t detected in current workspace \n[Initialize Operator Collection](command:operator-collection-sdk.initCollection)\n[skip](command:operator-collection-sdk.initCollectionSkip)",
```
Comment (freemanlatrell) [inline package.json:237]:
```suggestion
        "contents": "Operator Collection isn’t detected in current workspace \n[Initialize Operator Collection](command:operator-collection-sdk.initCollection)\n[Skip](command:operator-collection-sdk.initCollectionSkip)",
```

File: src/utilities/util.ts [hunk 2]
```diff
@@ -476,6 +487,40 @@ export async function requestLogInInfo(): Promise<string[] | undefined> {
...
+    prompt: `Enter your Ansible Galaxy namespace`,
+    ignoreFocusOut: true,
+    validateInput: text => {
+      const ocLoginArgs = text.trimStart();
+      const validValuesRegex = /^[a-zA-Z0-9]+$/;
+      const isvalid = !validValuesRegex.test(text?.trimStart());
+      return isvalid ? text : null;
+    },
+  });
+
```
Comment (freemanlatrell) [inline src/utilities/util.ts:507]: You should add validation check here to determine if the `ansibleGalaxyNamespace` is either `undefined` (meaning a user pressed Escape to cancel the prompts), or an empty string (meaning the user didn't specify the require value)

File: src/utilities/util.ts [hunk 3]
```diff
@@ -476,6 +487,40 @@ export async function requestLogInInfo(): Promise<string[] | undefined> {
...
+  const offlineInstall = await vscode.window.showInputBox({
+    prompt: `Will this collection be executed in an offline environment [y/n]`,
+    ignoreFocusOut: true,
+    validateInput: text => {
+      const validValuesRegex = /^(yes|y|no|n)$/i;
+      const isvalid = !validValuesRegex.test(text?.trimStart());
+      return isvalid ? text : null;
+    },
+  });
+
```
Comment (freemanlatrell) [inline src/utilities/util.ts:517]: Same validation here

**Review by freemanlatrell — CHANGES_REQUESTED** [review]
Also, how are you determining which directory to store the new initialized collection? When testing in a new VS Code window, it seems to be attempting to store this collection in my root directory....which causes the failure below:

```
TASK [Initialize collection] ***************************************************

fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["ansible-galaxy", "collection", "init", "undefined.racf_operator", "--init-path", "/"], "delta": "0:00:00.191989", "end": "2023-11-07 23:24:52.087264", "msg": "non-zero return code", "rc": 250, "start": "2023-11-07 23:24:51.895275", "stderr": "ERROR! Unexpected Exception, this is probably a bug: [Errno 30] Read-only file system: b'/undefined'", "stderr_lines": ["ERROR! Unexpected Exception, this is probably a bug: [Errno 30] Read-only file system: b'/undefined'"], "stdout": "to see the full traceback, use -vvv", "stdout_lines": ["to see the full traceback, use -vvv"]}
...ignoring

TASK [Create playbooks directory] **********************************************

fatal: [localhost]: FAILED! => {"changed": false, "msg": "There was an issue creating /undefined as requested: [Errno 30] Read-only file system: b'/undefined'", "path": "//undefined/racf_operator/playbooks/roles"}

PLAY RECAP *********************************************************************
localhost                  : ok=5    changed=2    unreachable=0    failed=1    skipped=1    rescued=0    ignored=1   
```

**Comment by MariandreaMV** [comment]
when opening a new window there is not folder opened, now It will prompt the user to open a new one as the following screenshot.
Screenshot 2023-11-08 at 4 17 43 PM

and if the folder doesn’t have a collection it will looks like this: 

Screenshot 2023-11-08 at 4 22 50 PM

**Review by freemanlatrell — APPROVED** [review]

**Pull request merged**